addon/components/bourbon-select-field.js

```diff
@@ -44,39 +45,46 @@ export default Component.extend(SelectMixin, {
   }),
 
   focusOut() {
+    this.set('activeOption', null);
     this.set('showList', false);
   },
 
   mouseDown() {
+    this.set('activeOption', null);
     this.set('showList', !this.get('showList'));
   },
 
   findValueObject(valueString) {
-    for (var optIndex in this.get('content')) {
-      if (this.get('content')[optIndex].value === valueString) {
+    for (var optIndex of this.get('content')) {
+      if (this.get('content')[optIndex].label === valueString) {
         return this.get('content')[optIndex];
       }
     }
+
   },
 
-  selection: computed('content.[]', 'value', {
+  keyDown(e) {
+    this.moveUpDown(e);
+    if (e.keyCode === 13) {
+      e.preventDefault();
+      this.send('updateSelection');
+      this.set('showList', false);
+      this.set('activeOption', null);
+      document.activeElement.blur();
+    }
+  },
+
+  selection: computed('value', {
     get(key) {
       this.getSelection();
     },
 
     set(key, value) {
       if (isPresent(value)) {
-        if (this.get('groupedContent') && this.get('value')) {
-          let valueString = this.get('value.label');
-          let searchValue = this.findValueObject(valueString);
-          this.setLabel(searchValue);
-          this.setValue(searchValue);
-        } else {
-          this.setLabel(value);
-          this.setValue(value);
-        }
+        this.setLabel(value);
+        this.setValue(value);
       }
-      return value;
```

Comment:
I think custom setters always need to return something.

Reply:
Ok. I added this back.  :)